### PR TITLE
add encodings sirun test

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
Adds a test of the encoders for each of the protocol version in the sirun benchmarks.

### Motivation

Encoding is a source of overhead.

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
I wonder if each directory should have a readme or some place for a description of the test scenario.

e.g. knowing that this test will call encode 10k times on a trace of 30 spans is a good tl;dr; if you don't want to read the code.

not a blocking comment, just thinking out loud about if/what/where we should consider this kind of description/documentation.

File: benchmark/sirun/encoding/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+for (let parent = null, i = 0; i < 30; i++) {
+  const span = createSpan(parent)
+  trace.push(span)
+  parent = span
+}
+
+const encoder = new AgentEncoder(writer)
+
+for (let j = 0; j < 10000; j++) {
+  encoder.encode(trace)
```
Comment (brettlangdon) [inline benchmark/sirun/encoding/index.js:46]: does the encoder maintain any kind of state between calls to encode?

I am thinking for v0.5 endpoint, does this scenario showcase the benefits we gain from that string table? (does it matter? is that a completely different scenario?)

Comment (bengl) [reply benchmark/sirun/encoding/index.js:46]: It should show that value. We already see differences in instruction counts between 0.4 and 0.5.

**Review by brettlangdon — APPROVED** [review]
I was thinking that maybe this test case is optimized in benefit of `0.5`, which is probably is a bit, but then a good majority of spans fall within some level of variation (e.g. two express requests might be 100% identical in all metadata except ids and timing, or maybe 98% identical because only resource and `http.url` are different, etc) so probably not worth trying to be overly pedantic right now on whether this is "real world representative", probably close enough to show some value

**Comment by bengl** [comment]
> I was thinking that maybe this test case is optimized in benefit of `0.5`

That's probably the case. That being said, I think this is sufficient for now. We can vary it later.

**Pull request merged**